dspace-api/src/main/java/org/dspace/app/util/DCInputsReader.java

```diff
@@ -424,25 +416,35 @@ private void processField(String formName, Node n, Map<String, String> field)
                             field.put(PAIR_TYPE_NAME, pairTypeName);
                         }
                     }
+                } else if (StringUtils.equalsIgnoreCase(tagName, "linked-metadata-field")) {
+                    for (int j = 0; j < nd.getChildNodes().getLength(); j ++) {
+                        Node nestedNode = nd.getChildNodes().item(j);
+                        String nestedTagName = nestedNode.getNodeName();
+                        String nestedValue = getValue(nestedNode);
+                        field.put(nestedTagName, nestedValue);
+                        if (nestedTagName.equals("input-type")) {
+                            handleInputTypeTagName(formName, field, nestedNode, nestedValue);
+                        }
+                    }
                 }
             }
         }
         String missing = null;
-        if (field.get("dc-element") == null) {
+        if (field.get("dc-element") == null && field.containsKey("field")) {
```

Comment:
Where does the `"field"` key gets added to the map? Is this the key of the whole entry in the `submission-forms.xml`? So maybe `n.tagName`? But this would not work for the `linked-metadata-field` case.

Reply:
You're right, the `"field"` key doesn't get added, it's name of the node instead. It also doesn't work for the `linked-metadata-field` case.
With the dtd in place, invalid metadata was not verified.

This has been resolved now and verified to break if the mandatory fields are missing

Reply:
This looks to be resolved now.  Marking resolved